WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -516,46 +522,32 @@ private extension OrderDetailsViewController {
     }
 
     @objc private func collectPayment(at: IndexPath) {
-        guard let cell = tableView.cellForRow(at: at) as? ButtonTableViewCell else {
-            return
-        }
-        /// TODO. Initiate UI flow for collecting payments
-        /// This will be removed later on, whenever we implement the proper UI flow
-        /// for collecting payments.
-        /// For now, we present an alert with a success/error message after completion.
-        /// This API does not have to be final. It is the simplest thing that
-        /// could possibly work at this point in time
-        viewModel.collectPayment { [weak self] readerEventMessage in
-            let title = "💳💳"
-            let message = readerEventMessage
-            let buttonTitle = "😊"
-
-            let actionSheet = UIAlertController(title: title, message: message, preferredStyle: .alert)
-            actionSheet.view.tintColor = .text
-
-            actionSheet.addCancelActionWithTitle(buttonTitle)
-
-            let popoverController = actionSheet.popoverPresentationController
-            popoverController?.sourceView = cell.contentView
+        paymentAlerts.readerIsReady(from: self,
+                                    name: viewModel.collectPaymentFrom,
```

Comment:
I've hedged about whether to suggest this above, but I'd almost rather the argument be `title` and not `name` both to match the internal calls (to Fancy Alerts, which I appreciate we will remove) but also to avoid me having to think "is this the buyer's name here?"